streams/src/test/java/org/apache/kafka/test/StreamsTestUtils.java

```diff
@@ -78,4 +72,24 @@ public static Properties minimalStreamsConfig() {
         return results;
     }
 
+    // TODO: these three static functions are added because some non-TopologyBuilder unit tests need to access the internal topology builder,
+    //       which is usually a bad sign of design patterns between TopologyBuilder and StreamThread. We need to consider getting rid of them later
+    public static InternalTopologyBuilder internalTopologyBuilder(final StreamsBuilder streamsBuilder) throws NoSuchFieldException, IllegalAccessException {
```

Comment:
Do we really need to use reflection? Can we not add a test class in the same package that exposes a package private field as public? When exploring the codebase is very useful to be able to see callers and reflection kills that.